gammapy/irf/irf_reduce.py

```diff
@@ -0,0 +1,154 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+from . import PSF3D, EnergyDependentTablePSF, IRFStacker
+from ..utils.energy import Energy
+
+__all__ = ["make_psf", "make_mean_psf", "make_mean_edisp"]
+
+
+def make_psf(self, position, energy=None, rad=None):
```

Comment:
Please rename `self` to `obs` here, and to add it to the `Parameters` list in the docstring.

`self` is  just for methods on classes, shouldn't be used for standalone functions.

Same change for the functions below, just in the last one call it `observations`.
(well, that's my preference, that we go for `observations` everywhere, currently we have a mix of `observations` and `obs_list` in Gammapy).

Reply:
Ah, sorry, I missed your comment that you want to do that in a follow-up PR.
Yes, good, of course, remove the methods on the obs classes and use these ones.